openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js

```diff
@@ -115,11 +115,29 @@ export default {
       }
     })
     if (this.screen) {
-      this.items.map((item) =>
+      this.items = this.items.map((item) => {
+        let itemName = item.itemName
+        // Remove double bracket escaping. This means they actually have an item
+        // with a bracket in the name, not an array index.
+        if (item.itemName.includes('[[')) {
+          // Change the actual item and the the name we're passing to the screen
+          item.itemName = item.itemName.replace('[[', '[').replace(']]', ']')
+          itemName = item.itemName
+        } else if (item.itemName.includes('[')) {
+          // Brackets mean array indexes (normally, but see above)
+          let match = item.itemName.match(/\[(\d+)\]/)
+          this.arrayIndex = parseInt(match[1])
+          // Here we keep the original item name with the brackets but change
+          // the item we're passing to the screen to the non-bracket item
+          itemName = item.itemName.replace(match[0], '')
+        }
         this.screen.addItem(
-          `${item.targetName}__${item.packetName}__${item.itemName}__${item.valueType}`,
-        ),
-      )
+          `${item.targetName}__${item.packetName}__${itemName}__${item.valueType}`,
+          true,
+        )
+        // Return the mapped values since we may have removed bracket escaping
+        return item
+      })
```

Comment:
Key to getting this work was reassigning this.items to the mapped version which returns the escaped bracket names

Reply:
This doesn't support an array item with a real bracket in the name.  That's probably ok.

Reply:
If you're naming an array item with brackets then you're really asking for trouble!